crates/ruff/src/rules/flake8_bugbear/rules/mutable_argument_default.rs

```diff
@@ -78,9 +90,44 @@ pub(crate) fn mutable_argument_default(checker: &mut Checker, arguments: &Argume
                 is_immutable_annotation(expr, checker.semantic())
             })
         {
-            checker
-                .diagnostics
-                .push(Diagnostic::new(MutableArgumentDefault, default.range()));
+            let mut diagnostic = Diagnostic::new(MutableArgumentDefault, default.range());
+
+            if checker.patch(diagnostic.kind.rule())
+                // Do not try to fix if the function is only one line.
+                && checker
+                    .locator
+                    .contains_line_break(TextRange::new(arguments.range().end(), body[0].start()))
+            {
+                // Set the default arg value to None
+                let arg_edit = Edit::range_replacement("None".to_string(), default.range());
+
+                // Add conditional check to set the default arg to its original value if still None
+                let mut check_lines = String::new();
+                let indentation = checker.locator.slice(TextRange::new(
+                    checker.locator.line_start(body[0].start()),
+                    body[0].start(),
+                ));
+                let indentation = leading_space(indentation);
+                // body[0].start() starts at correct indentation so we do need to add indentation
+                // before pushing the if statement
+                check_lines.push_str(format!("if {} is None:\n", def.arg.as_str()).as_str());
+                check_lines.push_str(indentation);
+                check_lines.push_str(checker.stylist.indentation());
+                check_lines.push_str(
+                    format!(
+                        "{} = {}",
+                        def.arg.as_str(),
+                        checker.generator().expr(default),
+                    )
+                    .as_str(),
+                );
+                check_lines.push_str(&checker.stylist.line_ending());
+                check_lines.push_str(indentation);
+                let check_edit = Edit::insertion(check_lines, body[0].start());
+
+                diagnostic.set_fix(Fix::manual_edits(arg_edit, [check_edit]));
```

Comment:
Manual edits won't be applied when running `--fix` in the future and are intended to be used for fixes that are known to contain syntax errors. Do you expect the fixes to contain syntax errors? If not, consider using `suggested` instead.

Reply:
Currently only syntax errors I am aware of that can occur is in the case of weirdly formatted single line funcs. See also: https://github.com/astral-sh/ruff/pull/4880#discussion_r1257490739 if that can be fixed it can go to suggested AFAIAC.